velocity/src/main/java/co/aikar/commands/VelocityCommandCompletions.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2016-2017 Daniel Ennis (Aikar) - MIT License
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining
+ *  a copy of this software and associated documentation files (the
+ *  "Software"), to deal in the Software without restriction, including
+ *  without limitation the rights to use, copy, modify, merge, publish,
+ *  distribute, sublicense, and/or sell copies of the Software, and to
+ *  permit persons to whom the Software is furnished to do so, subject to
+ *  the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be
+ *  included in all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ *  LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ *  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ *  WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package co.aikar.commands;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.velocitypowered.api.command.CommandSource;
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.ProxyServer;
+
+import co.aikar.commands.apachecommonslang.ApacheCommonsLangUtil;
+import net.kyori.text.format.TextColor;
+
+public class VelocityCommandCompletions extends CommandCompletions<VelocityCommandCompletionContext> {
+
+    public VelocityCommandCompletions(ProxyServer server, CommandManager manager) {
+        super(manager);
+        registerAsyncCompletion("chatcolors", c -> {
+            Stream<TextColor> colors = Stream.of(TextColor.values());
+            if (c.hasConfig("colorsonly")) {
+                colors = colors.filter(color -> color.ordinal() <= 0xF);
```

Comment:
This is useless. `TextColor` only has colours, no decorations - decorations are in `TextDecoration`.

Reply:
...and likewise, `TextDecoration` only has decorations, not colours.